### PR TITLE
Reland: Improve time to development by initializing frontend_server concurrently with platform build

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Initialize the frontend_server while we're waiting for the platform build to complete. Should slightly speed up flutter run. With this change I get real 0m20.547s for flutter run in the gallery on a macbook/ios simulator vs real 0m25.131s without.

Reland: Attempting to deflake integration test that otherwise failed.

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/hot_reload_test.dart [hunk 1]
```diff
@@ -36,10 +36,17 @@ void main() {
   });
 
   test('newly added code executes during hot reload', () async {
+    final StringBuffer stdout = StringBuffer();
+    final Completer<void> sawTick = Completer<void>();
+    final StreamSubscription<String> subscription = _flutter.stdout.listen((String line) {
+      if (line.contains('(((TICK 1)))')) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/integration.shard/hot_reload_test.dart]: Added similar logic to reassemble test to ensure we are 100% ready before we start the hot reload.

**Comment by jonahwilliams** [comment]
If all of the integration tests pass on the first go (which it looks like they will), then odds are the flakiness is resolved ... probably :)

File: packages/flutter_tools/test/integration.shard/hot_reload_test.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/test/integration.shard/hot_reload_test.dart]: Consider adding a comment here and below indicating what TICK 1 means.

**Review by zanderso — APPROVED** [review]
lgtm w/ comment suggestion.

File: packages/flutter_tools/test/integration.shard/hot_reload_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/integration.shard/hot_reload_test.dart]: Done

**Comment by jonahwilliams** [comment]
Well it failed on the second run, so it isn't fixed yet :cry:

**Comment by zanderso** [comment]
@jonahwilliams what is the status of this PR?

**Comment by jonahwilliams** [comment]
Still working out the timing issues

**Comment by jonahwilliams** [comment]
Closing to remove from review queue

**Comment by zanderso** [comment]
This was looking like a nice improvement. Could you maybe add some more detail here about what the timing issue was? Was it only for tests, or did the timing issues entail real bugs?

**Comment by jonahwilliams** [comment]
I"m still planning to take a look at this, but it seems to have uncovered a lot of timing issues that I need to resolve. Its not clear whether this is tickling an existing bug in the integration tests, or whether there are other problems with our initial timestamps used (they are too early/late)

**Pull request closed without merging**